### PR TITLE
fix(usdt-approval): :bug: fix swap zero approvals ignoring buy flow

### LINKED ISSUE #2521: [1.38]No warning banner for 2-step approval of USDT token for a buy order on the Swap page
# To Test

1. Go to CoW Swap on Goerli
2. Use USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Go to swap
5. Specify a Buy order: Enter an amount into the Buy field that makes USDT amount  higher than the approval amount (for example 10 USDT)

**AR**: No warning banner for 2-step approval of USDT token for a buy order, and no 'Reset' modal

**ER:** You should see the flow the warning and the Reset allowance modal

### PR DESCRIPTION
**Pull request by tukantje**
# Summary

Fixes #2521

Swap specific zero approval checker was not relying on slippage adjusted amounts. Updated that part of the logic to make it work for both sides of the flow.

Screenshot 2023-05-23 at 18 57 18
Screenshot 2023-05-23 at 18 57 11

  # To Test

1. Go to CoW Swap on Goerli
2. Use USDT https://goerli.etherscan.io/address/0x7b77F953e703E80CD97F6911385c0b1ceabC96Bc
3. Approve or trade to have some left over approvals (for example 5 USDT)
4. Go to swap
5. Specify a Buy order: Enter an amount into the Buy field that makes USDT amount higher than the approval amount (for example 10 USDT)
6. You should see zero approval warning and flow

### CODE REVIEW

File: src/common/hooks/useShouldZeroApproveSwap.ts [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
 import { useShouldZeroApprove } from './useShouldZeroApprove'
+import { computeSlippageAdjustedAmounts } from 'legacy/utils/prices'
 
 export function useShouldZeroApproveSwap() {
-  const { parsedAmount } = useDerivedSwapInfo()
-  return useShouldZeroApprove(parsedAmount)
+  const { v2Trade, allowedSlippage } = useDerivedSwapInfo()
+
+  const slippageAdjustedAmounts = computeSlippageAdjustedAmounts(v2Trade, allowedSlippage)
+  return useShouldZeroApprove(slippageAdjustedAmounts.INPUT)
```
Comment (alfetopito) [inline src/common/hooks/useShouldZeroApproveSwap.ts:9]: You don't need to compute both.

Example https://github.com/cowprotocol/cowswap/blob/57d43e2fb5e40bf1695cb6b8c7037cfdbd351640/src/legacy/components/swap/ConfirmSwapModal/index.tsx#L74

Just use:

```suggestion
  return useShouldZeroApprove(v2Trade.maximumAmountIn(allowedSlippage))
```

Comment (tukantje) [reply src/common/hooks/useShouldZeroApproveSwap.ts]: Good point - updated the PR. Thank you.

**Review by anxolin — APPROVED** [review]
Great Fix! And great catch by @elena-zh

**Comment by anxolin** [comment]
Remember to add QA into the PR

**Comment by alfetopito** [comment]
> Remember to add QA into the PR

It was there, but since you are part of the QA team, when you review, it counts as QA reviewed

**Review by shoom3301 — APPROVED** [review]

**Review by elena-zh — APPROVED** [review]
Great!

**Comment by anxolin** [comment]
> It was there, but since you are part of the QA team, when you review, it counts as QA reviewed

What, I don't like this. They added me cause I created the group. I will try to see if I can be removed

**Pull request merged**